src/coreclr/jit/gentree.h

```diff
@@ -4743,6 +4743,10 @@ struct GenTreeCall final : public GenTree
     unsigned char gtCallType : 3;   // value from the gtCallTypes enumeration
     unsigned char gtReturnType : 5; // exact return type
 
+#if defined(TARGET_WASM)
+    CorInfoType gtCallIrReturnType; // to help construct the LLVM signature for string ctors
```

Comment:
This is string constructors that return `string`, so for example I think it was
```
        string Ctor(char[] value, int startIndex, int length)
        {
```
Want me to get the exact IR/method ?

Reply:
> Want me to get the exact IR/method ?

No, I understand the string constructor situation, but for them - the IR node type would match this type (it would be `TYP_REF`). What I am asking is when `JITType2varType(gtCallIrReturnType) != call->TypeGet()`, i. e. when can we not use the node's type when the signature type disagrees with the node type.

Reply:
Sorry, I think I was confused.   Let me address https://github.com/dotnet/runtimelab/pull/1744#discussion_r757692693 first and then this might go away.